app/components/TermsCheckbox/index.tsx

```diff
@@ -0,0 +1,59 @@
+import { useIntl } from 'react-intl';
+import React, { useState } from 'react';
+import { Markup } from 'interweave';
+
+import { themeColors } from 'theme';
+import { Roles } from 'models/User/RolesManager';
+
+import FormikCheckbox from 'components/FormikCheckbox';
+import Text from 'components/Text';
+import Modal from 'components/Modal';
+
+import messages from './messages';
+import { TermsAndConditions } from './styled';
+
+type Props = {
+  role?: Roles;
+};
+
+const TermsCheckbox = ({ role }: Props) => {
+  const [showTermsModal, setShowTermsModal] = useState(false);
+  const { formatMessage } = useIntl();
+
+  const termsAndConditionsText =
+    role === Roles.Participant || !role
+      ? formatMessage(messages.termsAndConditionsParticipantText)
+      : formatMessage(messages.termsAndConditionsOtherRolesText);
+
+  return (
+    <>
+      <Modal
+        visible={showTermsModal}
+        title={formatMessage(messages.termsAndConditions)}
+        onClose={() => setShowTermsModal(false)}
+        maxWidth="80%"
+      >
+        <TermsAndConditions>
+          <Markup content={termsAndConditionsText} noWrap />
+        </TermsAndConditions>
+      </Modal>
+      <FormikCheckbox formikKey="terms">
```

Comment:
I suggest adding `formikKey` prop to `TermsCheckbox` component and using it here instead of hardcoded value because not only the key depends on usage but also you need to dive into the components code to see what is the key and adjust formik's schema to it.